dev/docker/README.md

```diff
@@ -0,0 +1,38 @@
+### Developing with Docker
```

Comment:
As part of Issue #901 - I'd like to see this moved into `docs/source/guides/docker-development-guide` or similar.